zathura/database-sqlite.c

```diff
@@ -242,6 +243,26 @@ sqlite_db_init(ZathuraSQLDatabase* db, const char* path)
     }
   }
 
+  /* check existing tables for correct column types */
+  ret1 = check_column_type(session, "fileinfo", "time", "TEXT", &res1);
+
+  if (ret1 == true && res1 == false) {
+    girara_debug("old database table layout detected; updating ...");
+
+    char SQL_FILEINFO_ALTER4[525] = "BEGIN TRANSACTION;"
```

Comment:
Please don't hardcode string sizes. We will certainly forget to update the size if one of the statements changes in the future. Either calculate it from the three parts or generate the final string dynamically.